testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/InstrumentationExtension.java

```diff
@@ -76,6 +80,13 @@ public List<MetricData> metrics() {
     return testRunner.getExportedMetrics();
   }
 
+  /** Returns a list of metrics, filtered by instrumentation name */
+  public List<MetricData> instrumentationMetrics(String instrumentationName) {
```

Comment:
We don't have analogous methods for any other signal; and it's only used in this class. WDYT about making it private?
```suggestion
  private List<MetricData> instrumentationMetrics(String instrumentationName) {
```

Reply:
I intended it to be a public helper, analogous to `metrics()` but filtered to a specific instrumentation. It's only used internally now, but I could see it being useful elsewhere. If you think it's too much YAGNI then I will make it private.

Reply:
Let's make it private -- there's just one use case for that now; and I think all 3 signals should have very similar API, so if we're introducing some new thing here we should do similar thing everywhere.

Reply:
Ok I relent.  :) I think you have a higher standard for test code than I do.

Reply:
Hah, I'm just thinking that YAGNI (probably) when it comes to making that method public